jax/_src/numpy/ufuncs.py

```diff
@@ -3061,6 +3064,17 @@ def frexp(x: ArrayLike, /) -> tuple[Array, Array]:
   return _where(cond, x, x1), lax.convert_element_type(x2, np.int32)
 
 
+@_frexp.defjvp
+def _frexp_jvp(primals, tangents):
+  x, = primals
+  t, = tangents
+  m, e = frexp(x)
+  # mdot = ldexp(t, -e)
+  mdot = t * exp2(-e)
```

Comment:
We maybe should use `ldexp` here, as it's more accurate for large values of `e`. However that results in failing this assertion:
```
  File "jax/_src/lax/lax.py", line 7310, in _select_transpose_rule
    assert not ad.is_undefined_primal(which)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

Reply:
Actually, maybe I shouldn't worry about this: the only problematic cases are when `exp2(-e)` overflows. but by construction, `e` is the exponent of a non-overflowing value, so it should be in range.